src/display.rs

```diff
@@ -357,7 +361,14 @@ where
 {
     let mut encoder = BufEncoder::<N>::new();
     encoder.put_bytes(bytes, case);
-    f.pad_integral(true, "0x", encoder.as_str())
+    let encoded = encoder.as_str();
+
+    if let Some(precision) = f.precision() {
+        if encoded.len() > precision {
+            return f.pad_integral(true, "0x", &encoded[..precision]);
```

Comment:
Rather than encoding the whole string and then truncating it'd be better to only encode up to `precision`.